core/src/test/scala/org/apache/spark/util/UninterruptibleThreadSuite.scala

```diff
@@ -115,6 +116,46 @@ class UninterruptibleThreadSuite extends SparkFunSuite {
     assert(interruptStatusBeforeExit)
   }
 
+  test("no runUninterruptibly") {
+    @volatile var hasInterruptedException = false
+    val latch = new CountDownLatch(1)
+    val t = new UninterruptibleThread("test") {
+      override def run(): Unit = {
+        latch.countDown()
+        hasInterruptedException = sleep(1)
+      }
+    }
+    t.start()
+    latch.await(10, TimeUnit.SECONDS)
+    t.interrupt()
```

Comment:
It looks that Java 8 behaves differently when `interrupt()` is called on not started thread.